app/assets/javascripts/components/sampleUpload.jsx

```diff
@@ -778,6 +803,32 @@ class SampleUpload extends React.Component {
                     </div> :
                     null
                 }
+                <div className='field'>
+                  <div className='row'>
+                    <div className='col no-padding s12'>
+                      <div className='field-title'>
+                        <div className='read-count-label'>
+                          Sample name
+                        </div>
+                      </div>
+                    </div>
+                  </div>
+                  <div className='row input-row'>
+                    <div className='col no-padding s12'>
+                      <input type='text' ref='sample_name' className='browser-default' value={this.state.generatedSampleName} onChange={ (e) => {
+                        this.setState({
+                            generatedSampleName: e.target.value.trim()
```

Comment:
I think you should also set a state so if people change Read path, the sample name  entered by the user won't be  overwritten. Also I think the function should be extracted out.